tests/api.c

```diff
@@ -38990,6 +38990,47 @@ static void test_wolfSSL_CRYPTO_memcmp(void)
  | wolfCrypt ASN
  *----------------------------------------------------------------------------*/
 
+static void test_wc_CreateEncryptedPKCS8Key(void)
+{
+#if defined(HAVE_PKCS8) && !defined(NO_PWDBASED)
+    WC_RNG rng;
+    byte* encKey = NULL;
+    word32 encKeySz = 0;
+    word32 decKeySz = 0;
+    const char password[] = "wolfSSL";
+    word32 passwordSz = XSTRLEN(password);
+    word32 tradIdx = 0;
+
+    printf(testingFmt, "test_wc_CreateEncryptedPKCS8Key");
+
+    AssertIntEQ(wc_InitRng(&rng), 0);
+    /* Call with NULL for out buffer to get necessary length. */
+    AssertIntEQ(wc_CreateEncryptedPKCS8Key((byte*)server_key_der_2048,
+        sizeof_server_key_der_2048, NULL, &encKeySz, password, passwordSz,
+        PKCS5, PBES2, AES256CBCb, NULL, 0, WC_PKCS12_ITT_DEFAULT, &rng, NULL),
+        LENGTH_ONLY_E);
+    AssertNotNull(encKey = (byte*)XMALLOC(encKeySz, NULL,
```

Comment:
XMALLOC/XFREE need `HEAP_HINT`.

Reply:
Isn't the heap hint `DYNAMIC_TYPE_TMP_BUFFER`? I'm using that here. But maybe we're talking about different things.